### PR TITLE
Fix cases where the system might be on sda and fix some shellcheck warnings

### PR DESCRIPTION
**Pull request by Thaodan**
- Use test/[] in consistent way.
- Fix warnings about using non POSIX features
- Use grep directly and quote.
- Loop by using globs instead of ls
- Test for cases where the disk is on sda
- Silence grep output with -q and ignore stderr

### CODE REVIEW

**Comment by rinigus** [comment]
In this form it didn't work on Akari. Will test 

```
while [ ! -d /sys/class/block/mmcblk0p* ] || [ ! -d /sys/class/block/sd* ] ; do
    echo "find-mmc-bypartlabel: Waiting for /sys/class/block/mmcblk0p*..." > /dev/kmsg
    i=$(( ${i:-0} + 1 ))
    sleep 0.5
    if [ $i = 10 ]; then
        echo "find-mmc-bypartlabel: Error: timeout waiting for /sys/class/block/mmcblk0p*" > /dev/kmsg
        exit 1
    fi
done

for mmc_sysfs in /sys/class/block/mmcblk0p*/*/ /sys/class/block/sd*/*/; do
        if grep -w PARTNAME="$1" "$mmc_sysfs"/uevent > /dev/null; then

```

next time.

**Comment by Thaodan** [comment]
The error was elsewhere, please test again as mentioned on IRC.

File: find-mmc-bypartlabel [hunk 1]
```diff
@@ -20,28 +20,28 @@
...
-    if [ $i == 10 ]; then
+    if [ $i = 10 ]; then
         echo "find-mmc-bypartlabel: Error: timeout waiting for /sys/class/block/mmcblk0p*" > /dev/kmsg
         exit 1
     fi
 done
 
-for mmc_sysfs in $(ls -d /sys/class/block/mmcblk0p*); do
-	if cat $mmc_sysfs/uevent | grep -w PARTNAME=$1 > /dev/null; then
+for mmc_sysfs in /sys/class/block/mmcblk0p*/*/ /sys/class/block/sda/*/; do
```
Comment (rinigus) [inline find-mmc-bypartlabel]: we need partition names over here: `sd*/*/`

Comment (Thaodan) [reply find-mmc-bypartlabel]: But the script ignores everything after mmcblk0* why should it do not for everything after sda*?

Comment (Thaodan) [reply find-mmc-bypartlabel]: Fixed that the error was that it was sda/*/ which means all subdir of sda and not sda*/ all dir in this folder beginning with sda.

**Comment by rinigus** [comment]
@Thaodan , it still failed. I suspect that it doesn't like to test with grep non-existing mmcblk0p* . works in terminal, not in initrd - maybe difference in shells. will be able to continue to test tomorrow night

File: find-mmc-bypartlabel [hunk 2]
```diff
@@ -20,28 +20,28 @@
...
 fi
 
-if ! test -d /sys/class/block; then
+if [ ! -d /sys/class/block ]; then
 	echo "$0: Error, please mount sysfs" > /dev/kmsg
 	exit 1
 fi
 
-while [ ! -d /sys/class/block/mmcblk0p* ]; do
+while [ ! -d /sys/class/block/mmcblk0p* ] && [ ! -d /sys/class/block/sda ] ; do
```
Comment (rinigus) [inline find-mmc-bypartlabel:33]: when I use 
```
while [ ! -d /sys/class/block/mmcblk0p* ] && [ ! -d /sys/class/block/sd* ] ; do
```

I get "line 33: [: too many arguments" when running in booted shell. I suggest that we should make a clean script without any errors like it. possibly original one was doing the same, though.

**Comment by rinigus** [comment]
Sorry for late reply, was fixing other issues with the port. Now when many things work, would like to clean all hacks away

**Comment by Thaodan** [comment]
Thats normal the test should 
just check if one of the files 
exist.

**Comment by rinigus** [comment]
@Thaodan : when taking your version as-it-is, Xperia XZ2 booted to SFOS. So, that's good.

Don't like, though, that it assumes /dev/sda. I wonder if /dev/sdb is possible as well, for example.

**Comment by Thaodan** [comment]
Its always the first device just as with the check without this PR the second is the sdcard usally so I wouldn't make sense.

**Comment by rinigus** [comment]
on xz2, we have sdb and sdc on internal storage. not the main data partition, though

**Comment by Thaodan** [comment]
Which partitions? 
Can you do an lsblk?

**Comment by rinigus** [comment]
See https://github.com/sailfishos-sony-tama/hybris-boot/blob/tama-akari/fixup-mountpoints#L238 and around

**Comment by rinigus** [comment]
Any news regarding this? Would someone mind to review and merge this?

**Comment by mkosola** [comment]
I tested this and seem to work nicely. Could you still change a bit your commit messages so that we get those in changelog: https://sailfishos.org/wiki/Building_packages#Changelog_generation Also one MER bug is needed.

**Comment by rinigus** [comment]
Great, thanks for testing! @Thaodan - so ball is on your side now :)

**Comment by Thaodan** [comment]
Bug is created: https://bugs.merproject.org/show_bug.cgi?id=2073
I added contributes to MER#2073 to every commit plus the tag as in other changes made by Jolla recently.
I used contributes instead of fixes as the sum of changes fixes the issue. In the merge message there should be a mention of: Fixes MER#2073.

**Review by mkosola — APPROVED** [review]

**Comment by mkosola** [comment]
Thank you, will merge and tag!

**Pull request merged**